vector/src/main/java/im/vector/util/EventGroup.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2017 Vector Creations Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.util;
+
+import android.content.Context;
+import android.text.TextUtils;
+
+import org.matrix.androidsdk.adapters.MessageRow;
+import org.matrix.androidsdk.call.MXCallsManager;
+import org.matrix.androidsdk.rest.model.Event;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import im.vector.R;
+import im.vector.adapters.AdapterUtils;
+
+/**
+ * A EventGroup is a special event that can contain MessageRows
+ */
+public class EventGroup extends Event {
+
+    // events rows map
+    private final Map<String, MessageRow> mRowsMap;
+
+    // rows list (ordered)
+    private final List<MessageRow> mRows;
+
```

Comment:
because there is an addToFront method (it is not possible to insert at a dedicated position in a linkedHashMap)